### PR TITLE
Expand ProtocolErrorCode enums

### PR DESCRIPTION
**Pull request by jeffk-kiva**
| 🔥 | 🐞 | 🙋 | 🚫 | 🚀 |
|----|----|----|----|----|
|       |       |      |       |       |

*([Click here](https://github.com/kiva/protocol/blob/master/PULL_REQUEST_README.md) if you don't understand these emojis)*

**What issue is this targeting?**

**Changes proposed in this pull request**
* Add a bunch of enums to ProtocolErrorCode.
* Style cleanup:
  * Add a tslint rule to enforce standard spacing between brackets in import statements.
  * Set tsconfig to include a noUnusedLocals rule to highlight unused imports and local variables.
  * Files should end with an empty line
  * Indenting was inconsistent
  * Removed author attribution since in reality all of these files are worked on by many authors
  * Changed some comments into JSDocs to make life a little better for those of us working from an IDE

**🚀 Deployment changes 🚀**  
(_list added or removed env, db changes etc_)  
Increases the version to 1.22

**Other Notes**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
+    "module": "commonjs",
+    "declaration": true,
+    "noImplicitAny": false,
+    "removeComments": true,
+    "noLib": false,
+    "allowSyntheticDefaultImports": true,
+    "emitDecoratorMetadata": true,
+    "experimentalDecorators": true,
+    "sourceMap": true,
+    "noUnusedLocals": true,
```
Comment (jeffk-kiva) [inline tsconfig.json:12]: This is the only actual addition in this file. The rest of the changes are from making indenting consistent.

File: tslint.json [hunk 2]
```diff
@@ -36,25 +34,16 @@
...
         ],
-        "semicolon": [
-            true
-        ],
-        "variable-name": [
-            false
-        ]
+        "semicolon": true,
+        "variable-name": false,
+        "whitespace": [true, "check-module"]
```
Comment (jeffk-kiva) [inline tslint.json:46]: This is the only actual addition in this file. The rest of the changes are from making spacing consistent.

File: test/package.e2e-spec.ts [hunk 3]
```diff
@@ -73,51 +72,50 @@
...
+        } catch (e) {
+            exceptionFound = true;
+        }
+
+        expect(exceptionFound).toBe(true);
+        return;
     });
 
-    describe('Retry logic tests', () => {
-        it('retry throws exception on timeout', async () => {
```
Comment (matt-raffel-kiva) [inline test/package.e2e-spec.ts:79]: This test needs to remain.  I think the confusing is we have 2 tests named the same but they are not actually the same.

**Review by matt-raffel-kiva — CHANGES_REQUESTED** [review]

File: test/package.e2e-spec.ts (hunk 3 above)
Comment (jeffk-kiva) [reply test/package.e2e-spec.ts:79]: Ahh I didn't actually delete anything in this file, it's just spacing changes and removing a couple unused local variables in the `beforeAll` block. The way github shows changes is weird.

Comment (matt-raffel-kiva) [reply test/package.e2e-spec.ts:79]: my apologizes.  I thought it was strange. lgtm

**Review by matt-raffel-kiva — APPROVED** [review]

**Pull request merged**